File: Tokopedia/Classses/UIRefactor/Home/MyWishlist/MyWishlistResponse.m
//
//  MyWishlistResponse.m
//  Tokopedia
//
//  Created by Billion Goenawan on 9/23/16.
//  Copyright © 2016 TOKOPEDIA. All rights reserved.
//

#import "MyWishlistResponse.h"
#import "MyWishlistData.h"
#import "Paging.h"

@implementation MyWishlistResponse


+(RKObjectMapping*)mapping
{
    RKObjectMapping *mapping = [RKObjectMapping mappingForClass:self];
    [mapping addPropertyMapping:[RKRelationshipMapping relationshipMappingFromKeyPath:@"data" toKeyPath:@"data" withMapping:[MyWishlistData mapping]]];
    [mapping addPropertyMapping: [RKRelationshipMapping relationshipMappingFromKeyPath:@"pagination" toKeyPath:@"pagination" withMapping:[Paging mappingForWishlist]]];
    return mapping;
}
@end

File: Tokopedia/Classses/UIRefactor/Home/MyWishlist/MyWishlistResponse.swift
//
//  MyWishlistResponse-Swift.swift
//  Tokopedia
//
//  Created by Billion Goenawan on 10/14/16.
//  Copyright © 2016 TOKOPEDIA. All rights reserved.
//

import UIKit

@objc(MyWishlistResponse)
class MyWishlistResponse: NSObject {
    
    var data: [MyWishlistData]!
    var pagination: Paging!
    var header: GeneralMetaData!
    
    class func mapping() -> RKObjectMapping {
        let mapping = RKObjectMapping(for: MyWishlistResponse.self)
        mapping!.addPropertyMapping(RKRelationshipMapping(fromKeyPath: "data", toKeyPath: "data", with: MyWishlistData.mapping()))
        mapping!.addPropertyMapping(RKRelationshipMapping(fromKeyPath: "pagination", toKeyPath: "pagination", with: Paging.mappingForWishlist()))
        mapping!.addPropertyMapping(RKRelationshipMapping(fromKeyPath: "header", toKeyPath: "header", with: GeneralMetaData.mapping()))
        return mapping!
    }
}
